### PR TITLE
Integrate Telemetry SDK as framework dependency

### PR DESCRIPTION
**Pull request by julianrex**
Test integration of Mapbox Mobile Events as a dependency. 

This PR adds `mapbox-events-ios` as a separate project to the Xcode workspace, and links with the `MapboxMobileEvents.framework`:
 
- CocoaPods: no change (events is specified as a dependency in the Podspec)
- Carthage: as an interim measure, please specify `github "mapbox/mapbox-events-ios" == 0.10.1-alpha` in your Cartfile. It is still TBD how the Carthage dependency will be specified.
- Manual installation: New Github release artifact (`mapbox-ios-sdk-5.6.0-alpha.1-dynamic-with-events.zip`) contains the `MapboxMobileEvents.framework`; please install this along with `Mapbox.framework`

This PR also bumps the release version to `5.6.0-alpha.1`. As with all alphas and betas, it is not recommended for production use.

----

OP:

Required for `alpha.1`:
- [x] Fix CI failures
- [x] @rclee review changes in `mapbox-events-ios`
- [x] Update `MGLMapboxEvents` with @rclee's help
- [x] Update packaging script 
- [x] Update CI to build/exercise new target

Optional for `alpha.1`:
- [X] Review decision to link statically with events for SDK's Github release
    - Decision to avoid linking statically with the events library for the immediate release (the current static framework currently needs work regardless)
- [ ] Add tests - this should be a separate PR that we can land independently of this PR. Check behavior hasn't changed.
- [ ] Document install process if necessary (e.g. if Github release requires 2 frameworks instead of 1)

### CODE REVIEW

**Comment by julianrex** [comment]
This PR bumps the no-content timeout to 20mins (up from 5 minutes) - we need to investigate this as tail work.

**Comment by julianrex** [comment]
Waiting on https://github.com/mapbox/mapbox-gl-native-ios/pull/67 to rebase.

File: platform/ios/ios.xcodeproj/xcshareddata/xcschemes/dynamic.xcscheme [hunk 1]
```diff
@@ -87,7 +87,7 @@
       </MacroExpansion>
    </LaunchAction>
    <ProfileAction
-      buildConfiguration = "Release"
+      buildConfiguration = "RelWithDebInfo"
```
Comment (fabian-guerra) [inline platform/ios/ios.xcodeproj/xcshareddata/xcschemes/dynamic.xcscheme]: Why is this changing?

File: platform/ios/src/MGLMapView+Impl.mm [hunk 2]
```diff
@@ -4,7 +4,8 @@
 #import "NSBundle+MGLAdditions.h"
 
 #if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
-#import "MMEEventsManager.h"
+#import <MapboxMobileEvents/MapboxMobileEvents.h>
+//#import "MMEEventsManager.h"
```
Comment (fabian-guerra) [inline platform/ios/src/MGLMapView+Impl.mm]: nit: please remove unused code

File: platform/ios/ios.xcodeproj/xcshareddata/xcschemes/dynamic.xcscheme (hunk 1 above)
Comment (julianrex) [reply platform/ios/ios.xcodeproj/xcshareddata/xcschemes/dynamic.xcscheme]: This is a hold-over from previous testing - however, it's easier to get meaningful results when profiling with symbols.

**Pull request merged**